geomesa-convert/geomesa-convert-common/src/main/scala/org/locationtech/geomesa/convert/Transformers.scala

```diff
@@ -482,5 +481,43 @@ class MapListFunctionFactory extends TransformerFunctionFactory {
         }.toMap.asJava
     }
   }
-
+}
+class CastFunctionFactory extends TransformerFunctionFactory {
+  override def functions = Seq(string2double,string2int,string2float,string2long,string2boolean)
+
+  val string2double = TransformerFn("string2double") {
+    args => {
+      if (args(0) == null || args(0).asInstanceOf[String] == "") {
+        if (args(1) == null || args(1).asInstanceOf[String] == "") args(1) else args(1).asInstanceOf[String].toDouble
```

Comment:
Not sure what you mean by that, can you clarify?  The only way you should be getting a non-double returned is if you explicitly provide one as a default value [i.e. args(1)](this should be clearer when I add the documentation, my bad).  Am I misunderstanding what you meant?